netman/dbus.go

```
@@ -135,104 +171,111 @@ func getSystemBus() *dbus.Conn {
-	devices := getDevices()
-	wifiDevices := getWifiDevices(devices)
+// Check if connected by wifi
+func (c *Client) ConnectedWifi(wifiDevices []string) bool {
 	for _, d := range wifiDevices {
 		objPath := dbus.ObjectPath(d)
-		device := conn.Object("org.freedesktop.NetworkManager", objPath)
-		device.Call("org.freedesktop.NetworkManager.Device.Disconnect", 0)
+		c.dbusClient.Object("org.freedesktop.NetworkManager", objPath)
+		setObject(c, "org.freedesktop.NetworkManager", objPath)
+		state, _ := c.dbusClient.BusObj.GetProperty("org.freedesktop.NetworkManager.Device.State")
+		if dbus.Variant.Value(state) == uint32(100) {
+			return true
+		}
 	}
-	return
+	return false
 }
 
-// SetIfaceManaged sets certain interface managed by netman
-func SetIfaceManaged(iface string) {
-	conn := getSystemBus()
-	devices := getDevices()
+// Disconnects every interface passed. return shows number of disconnect calls  made
+func (c *Client) DisconnectWifi(wifiDevices []string) int {
+	ran := 0
+	for _, d := range wifiDevices {
+		objPath := dbus.ObjectPath(d)
+		c.dbusClient.Object("org.freedesktop.NetworkManager", objPath)
+		setObject(c, "org.freedesktop.NetworkManager", objPath)
+		c.dbusClient.BusObj.Call("org.freedesktop.NetworkManager.Device.Disconnect", 0)
+		ran += 1
+	}
+	return ran
+}
 
+// Set passed device to be managed by network manager, return iface set, if any
+func (c *Client) SetIfaceManaged(iface string, devices []string) string {
+	ran := ""
 	for _, d := range devices {
 		objPath := dbus.ObjectPath(d)
-		device := conn.Object("org.freedesktop.NetworkManager", objPath)
-		deviceIface, err2 := device.GetProperty("org.freedesktop.NetworkManager.Device.Interface")
+		c.dbusClient.Object("org.freedesktop.NetworkManager", objPath)
+		setObject(c, "org.freedesktop.NetworkManager", objPath)
+		iface_, err2 := c.dbusClient.BusObj.GetProperty("org.freedesktop.NetworkManager.Device.Interface")
 		if err2 != nil {
-			fmt.Printf("Error in SetIfaceManaged(): %v\n", err2)
```

Maybe we can return the error also in this method, besides the interface set